saleor/graphql/warehouse/mutations.py

```diff
@@ -0,0 +1,113 @@
+import graphene
+from django.core.exceptions import ValidationError
+
+from ...shipping.models import ShippingZone
+from ...warehouse import models
+from ..account.i18n import I18nMixin
+from ..core.mutations import ModelDeleteMutation, ModelMutation
+from ..core.types.common import WarehouseError
+from .types import WarehouseCreateInput, WarehouseUpdateInput
+
+ADDRESS_FIELDS = [
+    "street_address_1",
+    "street_address_2",
+    "city",
+    "city_area",
+    "postal_code",
+    "country",
+    "country_area",
+    "phone",
+]
+
+
+class WarehouseMixin:
+    @classmethod
+    def validate_warehouse_count(cls, shipping_zones, instance):
+        """Every ShippingZone can be assigned to only one warehouse.
+
+        If not there would be issue with automatically selecting stock for operation.
+        """
+
+        warehouses = set(
```

Comment:
I am watching at it over 10 minutes and I am not able to understand how it works. Can you provide some description? HEre or in the doc string?

Reply:
BTW - is that method called automatically by graphene or have to be called explicit?

Reply:
@krzysztofwolski it must be called explicit (see `clean_input`)

Reply:
@szewczykmira lets move it outside of the mixin:
- much easier testing 
- it will be possible to use it outside of the mutations (what if I want to make some operations in management command?)

Reply:
@krzysztofwolski doesn't want to have any business logic inside mutations ;) 👍